libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1179,12 +1126,87 @@ export class RushConfiguration {
    * @deprecated Use `getCommittedShrinkwrapFilename` instead, which gets the correct common
    * shrinkwrap file name for a given active variant.
    */
-  public get committedShrinkwrapFilename(): string {
-    return this.getCommittedShrinkwrapFilename();
+  public committedShrinkwrapFilename(subspaceName?: string | undefined): string {
+    return this.getCommittedShrinkwrapFilename({ subspaceName });
   }
 
   /**
-   * Returns full path of the temporary shrinkwrap file for a specific subspace.
+   * The folder where Rush's additional config files are stored.  This folder is always a
+   * subfolder called `config\rush` inside the common folder.  (The `common\config` folder
+   * is reserved for configuration files used by other tools.)  To avoid confusion or mistakes,
+   * Rush will report an error if this this folder contains any unrecognized files.
+   *
+   * Example: `C:\MyRepo\common\config\rush`
+   */
+  public getCommonRushConfigFolder(subspaceName?: string | undefined): string {
+    if (subspaceName) {
+      this.validateSubspaceName(subspaceName);
```

Comment:
In the current impelmentation, `getCommonRushConfigFolder()` gets called repeatedly, and each time it re-validates the `subspaceName``.  That is a performance concern.

We should consider adding a simple cache to optimize these tests.  Even if the cache only remembers the one answer from the most recent `subspaceName` that was passed in, that would probably be a cache hit in most cases.

Reply:
Another idea would be to make a `Subspace` class that represents a validated subspace name and its precomputed associated paths. The API's would pass around this object instead of the `subspaceName` string, which would avoid the need to repeatedly call `rushConfiguration.getCommonTempFolder()` etc.

Reply:
Added a `_subspaceConfigFolderBySubspace` map that serves as a subspace config folder cache